position_in_partition.hh

```diff
@@ -132,6 +132,14 @@ public:
     position_in_partition_view(const clustering_key_prefix& ck, bound_weight w)
         : _type(partition_region::clustered), _bound_weight(w), _ck(&ck) { }
 
+    // delete constructors accepting clustering_key_prefix&&
+    // to help prevent cases where a view to a temporary
+    // clustering_key_prefix is used past the object's life time.
+    // (See #10555)
+    position_in_partition_view(clustering_row_tag_t, clustering_key_prefix&&) = delete;
+    position_in_partition_view(clustering_key_prefix&&) = delete;
+    position_in_partition_view(clustering_key_prefix&&, bound_weight) = delete;
```

Comment:
There's a `static position_in_partition_view for_key(const clustering_key& ck) {` thing that would accept temporary object. I think all `for_key`-s should either be removed, so that the p._i._p._view object is explicitly constructed from clustering_key reference, or added a `for_key(clustering_key&&)` overload, but ... how to make it fail the compilation if being used?

Reply:
> how to make it fail the compilation if being used?

Maybe add a `requires` concept that would always be false?